communication/src/dtls_message_channel.cpp

```diff
@@ -230,31 +258,16 @@ ProtocolError DTLSMessageChannel::init(
 	this->server_public = new uint8_t[server_public_len];
 	memcpy(this->server_public, server_public, server_public_len);
 	this->server_public_len = server_public_len;
+
+	// Records received from the server are not expected to contain a CID
+	mbedtls_ssl_conf_cid(&conf, 0 /* len */, MBEDTLS_SSL_UNEXPECTED_CID_IGNORE);
```

Comment:
Nit: `MBEDTLS_SSL_CID_IN_LEN_MAX` perhaps?

Reply:
I'd prefer to use 0 here as it makes the intent clear that we don't expect a CID in server-originated messages. `MBEDTLS_SSL_CID_IN_LEN_MAX` defined as 0 in our config is more a memory optimization.